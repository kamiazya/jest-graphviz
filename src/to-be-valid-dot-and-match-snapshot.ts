import { toMatchSnapshot } from 'jest-snapshot';
import { isValidDot } from './lib/is-valid-dot';

declare global {
  namespace jest {
    // tslint:disable-next-line: interface-name
    interface Expect {
      toBeValidDotAndMatchSnapshot<T>(): JestMatchers<T>;
    }
    // tslint:disable-next-line: interface-name
    interface Matchers<R, T> {
      toBeValidDotAndMatchSnapshot(): R;
    }
  }
}

export function toBeValidDotAndMatchSnapshotMatcher(this: any, dot: string): jest.CustomMatcherResult {
  const result = isValidDot(dot);
  if (result.pass) {
    return toMatchSnapshot.call(this, dot);
  }
  return result;
}

expect.extend({ toBeValidDotAndMatchSnapshot: toBeValidDotAndMatchSnapshotMatcher });
