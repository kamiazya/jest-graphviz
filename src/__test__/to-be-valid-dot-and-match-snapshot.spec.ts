import '../to-be-valid-dot-and-match-snapshot';

describe('toBeValidDotAndMatchSnapshot test', () => {
  test('matcher works', () => {
    const dot = 'digraph g { a -> b; }';
    expect(dot).toBeValidDotAndMatchSnapshot();
  });
});
