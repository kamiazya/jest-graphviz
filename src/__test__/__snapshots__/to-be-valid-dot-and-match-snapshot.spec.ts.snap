// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeValidDotAndMatchSnapshot test matcher works 1`] = `digraph g { a -> b; }`;
